### PR TITLE
EE-1047: Refactor bond/unbond outside core API.

### PR DESCRIPTION
**Pull request by mpapierski**
Ref: https://casperlabs.atlassian.net/browse/EE-1047

This PR:

- Refactors `bond`/`unbond` as internal impl detail of auction contract
- Cleans up a bit of test contracts (i.e. mint-bonding renamed with auction-bonding).
- Not calling bond/unbond directly from test contracts also required a bit of thoughful changes to the assertions
- Assumes bond/unbond is now renamed into `add_bid`/`withdraw_bid`, so some related past regression tests are updated for new entrypoints.

### CODE REVIEW

**Review by henrytill — APPROVED** [review]
Looking good.

File: grpc/tests/src/test/regression/ee_598.rs [hunk 1]
```diff
@@ -18,7 +18,7 @@ const ARG_PUBLIC_KEY: &str = "public_key";
 const ARG_ENTRY_POINT: &str = "entry_point";
 const ARG_ACCOUNT_HASH: &str = "account_hash";
 
-const CONTRACT_MINT_BONDING: &str = "mint_bonding.wasm";
+const CONTRACT_AUCTION_BONDING: &str = "auction_bonding.wasm";
```
Comment (EdHastingsCasperLabs) [inline grpc/tests/src/test/regression/ee_598.rs:21]: wouldn't this be auction_bidding.wasm now?

**Review by EdHastingsCasperLabs — APPROVED** [review]

File: grpc/tests/src/test/regression/ee_598.rs (hunk 1 above)
Comment (henrytill) [reply grpc/tests/src/test/regression/ee_598.rs:21]: Good point - we should be changing file/crate names here as well.

Comment (mpapierski) [reply grpc/tests/src/test/regression/ee_598.rs]: Fixed. While searching for `bond`/`unbond` I also got rid of a one regression test that was testing against `pos_bonding_purse` which doesn't make sense under auction.

**Pull request merged**